src/audio/pipeline/pipeline-schedule.c

```diff
@@ -91,6 +91,18 @@ static enum task_state pipeline_task_cmd(struct pipeline *p,
 		pipe_err(p, "pipeline_task_cmd(): failed to trigger components: %d", err);
 		reply->error = err;
 		err = SOF_TASK_STATE_COMPLETED;
+		switch (cmd) {
+		case COMP_TRIGGER_START:
+		case COMP_TRIGGER_RELEASE:
+		case COMP_TRIGGER_PRE_START:
+		case COMP_TRIGGER_PRE_RELEASE:
+			p->status = COMP_STATE_PAUSED;
```

Comment:
why is it PAUSED in all cases of failure @kv2019i ? What if the state was previously PREPARE before the trigger was run?

Reply:
@ranj063  If the state is not paused, pipeline_trigger() will assume pipeline task is running and send the IPC to be handled in the task. If start fails, there's no task, and IPC will never be responded to, leading to IPC timeout. But yup, maybe we need a another method to discover that the task is not running.

Reply:
> @ranj063 If the state is not paused, pipeline_trigger() will assume pipeline task is running and send the IPC to be handled in the task. If start fails, there's no task, and IPC will never be responded to, leading to IPC timeout. But yup, maybe we need a another method to discover that the task is not running.

Reporting non running task would have to be at the IPC level, as this sounds quite IPC specific.